pilot/pkg/security/authz/model/model.go

```diff
@@ -265,3 +270,23 @@ func (m *Model) Generate(service *ServiceMetadata, forTCPFilter bool) *envoy_rba
 	}
 	return policy
 }
+
+// getPrincipalsIncludingAliases returns list of users encoded in spiffe format from the local trust domain
+// and its aliases.
+// For example, if the local trust domain is "cluster.local", and its aliases are "td1", "td2".
+// For a user "bar" in namespace "foo", getPrincipalsIncludingAliases returns
+// cluster.local/ns/foo/sa/bar, td1/ns/foo/sa/bar, td2/ns/foo/sa/bar
+func getPrincipalsIncludingAliases(trustDomainAliases []string, principals []string) []string {
+	usersWithAliases := principals
+	for _, tdAlias := range trustDomainAliases {
+		for _, principal := range principals {
+			identityParts := strings.Split(principal, "/")
+			if len(identityParts) != 5 {
+				rbacLog.Errorf("Wrong SPIFFE format found: %s", principal)
```

Comment:
Do we have this validation at submitting time?

Also, may be better to return an empty list (or original list) if any of the principal is invalid, rather than the new partial list (which may already have few principals added).